api/src/db/models/opportunity_models.py

```diff
@@ -212,6 +212,14 @@ class OpportunityAssistanceListing(ApiSchemaTable, TimestampMixin):
 class LinkOpportunitySummaryFundingInstrument(ApiSchemaTable, TimestampMixin):
     __tablename__ = "link_opportunity_summary_funding_instrument"
 
+    __table_args__ = (
+        # We want a unique constraint so that legacy IDs are unique for a given opportunity summary
+        UniqueConstraint("opportunity_summary_id", "legacy_funding_instrument_id"),
+        # Need to define the table args like this to inherit whatever we set on the super table
+        # otherwise we end up overwriting things and Alembic remakes the whole table
+        ApiSchemaTable.__table_args__,
```

Comment:
Surprised this works, I would expect it to make a nested tuple and need a `*` to expand it into the top-level tuple:

```suggestion
        *ApiSchemaTable.__table_args__,
```

Reply:
That would make sense for how you structure it, but it isn't actually setup that way. https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#declarative-table-configuration

It outright errors with that format. The format when you have both tuples and a dictionary is this weird hybrid.